From 223817bdcdcf88efc243a22ef88dbbed765463ed Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@nabladev.com>
Date: Wed, 8 Oct 2025 13:41:36 +0200
Subject: [PATCH 93/95] drm/bridge: tc358767: Disable FRMSYNC

Disable FRMSYNC on the bridge chip, contrary to what TC9595 RM
recommends. This seems to stabilize the DP even at 1920x1080@60.
The current hypothesis is, that the STM32MP25xx DSI provides data
either too fast or too slow compared to the timings generated by
the TC9595.

Upstream-Status: Pending
Signed-off-by: Marek Vasut <marex@nabladev.com>
---
 drivers/gpu/drm/bridge/tc358767.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/bridge/tc358767.c b/drivers/gpu/drm/bridge/tc358767.c
index 1d31f8fad047e..993fa580bf555 100644
--- a/drivers/gpu/drm/bridge/tc358767.c
+++ b/drivers/gpu/drm/bridge/tc358767.c
@@ -925,7 +925,7 @@ static int tc_set_common_video_mode(struct tc_data *tc,
 	 */
 	ret = regmap_write(tc->regmap, VPCTRL0,
 			   FIELD_PREP(VSDELAY, right_margin + 10) |
-			   OPXLFMT_RGB888 | FRMSYNC_ENABLED | MSF_DISABLED);
+			   OPXLFMT_RGB888 | FRMSYNC_DISABLED | MSF_DISABLED);
 	if (ret)
 		return ret;
 
-- 
2.51.0

